Make tooltip content in popup screenreader-accessible

lenacohen opened the pull request:
**Addresses:**
- https://github.com/EFForg/privacybadger/issues/3138
- https://github.com/EFForg/privacybadger/issues/3132
- https://github.com/EFForg/privacybadger/issues/3131
- Made the collapsed/expanded state of the tracker list screenreader-accessible
- Added `role="radiogroup"` and an aria-label to the slider group to provide screenreaders with more context for slider buttons

Tested with the VoiceOver screenreader on MacOS

**Test instructions:**
- Navigate to each of the elements that have a tooltip:
  - EFF logo
  -  "trackers" link
  - Icons above the tracker list representing the slider options
  - Domains in the tracker list
  -  Domain slider options
  - Breakage warning tooltip
  - "Click to return control" tooltip
  - Google sign-in warning tooltip
  - DNT icon
- To see the breakage warning tooltip, move a slider in the yellow (cookie-block) position to the red (block) position
<img width="385" height="95" alt="Screenshot 2025-09-19 at 5 00 06 PM" src="https://github.com/user-attachments/assets/6c280513-60e0-46ea-a6bb-c1bd181ecc12" />

- To see the "Click to return control" tooltip, move a slider to any position
<img width="422" height="98" alt="Screenshot 2025-09-19 at 4 58 03 PM" src="https://github.com/user-attachments/assets/6ae2cb22-d9f4-43c3-956a-a42617947de2" />

- To see the "DNT" icon, visit https://www.nytimes.com/ and scroll down the tracker list to the domain with the DNT icon
<img width="364" height="73" alt="Screenshot 2025-09-19 at 5 02 37 PM" src="https://github.com/user-attachments/assets/396fd23f-b9e0-49ce-a690-1bff45e89f75" />

- To see the Google sign-in warning tooltip, visit https://stackoverflow.com/ or another site with Google sign-in
 <img width="436" height="187" alt="Screenshot 2025-09-19 at 4 56 31 PM" src="https://github.com/user-attachments/assets/9d80b8c8-60de-436d-84b4-55c118ca269b" />

ghostwords commented on src/js/htmlutils.js:
```diff
@@ -150,7 +150,7 @@ let htmlUtils = {
       return `
       <div class="dnt-compliant">
-        <a target=_blank href="https://privacybadger.org/#-I-am-an-online-advertising-tracking-company.--How-do-I-stop-Privacy-Badger-from-blocking-me"><img src="${dnt_icon_url}"></a>
+        <a target=_blank href="https://privacybadger.org/#-I-am-an-online-advertising-tracking-company.--How-do-I-stop-Privacy-Badger-from-blocking-me" aria-label="${i18n.getMessage('dnt_tooltip')}"><img src="${dnt_icon_url}" alt=""></a>
```
Let's cache this message here too, as we already cache `dnt_icon_url`.

ghostwords commented on src/js/popup.js:
```diff
@@ -571,6 +573,11 @@ function createBreakageNote(domain, i18n_message_key) {
+      // Add aria-hidden to ensure that the tooltip isn't read twice by screenreaders
+      // It should be read along with the selected slider option (via aria-describedby), and not where it's inserted at the end of the popup DOM
+      $(tooltip.elementTooltip()).attr({role: "tooltip", "aria-hidden": "true"});
+      $slider_allow .closest(".switch-toggle").find("input:checked").attr("aria-describedby", tooltip.elementTooltip().id);
```
Nit: space before the dot.

lenacohen commented on src/_locales/en_US/messages.json:
```diff
@@ -115,6 +115,16 @@
         "description": "Domain slider list header tooltip."
     },
+    "popup_slider_label": {
+        "message": "Blocking options for $DOMAIN$",
```
What about "Settings for $DOMAIN$" or "Blocking mode for $DOMAIN$"?

Other blocking extensions don't have sliders (except for UBlock's "filtering mode") and/or are not very screenreader accessible

lenacohen commented:
> We want want to have the same stuff happen in the options page. If we don't do it here, we might want to have the helper functions live in a common location so that we can do it on the options page when ready.
> 
> It doesn't look like the "Sign in with google" breakage warning is accounted for when keyboard navigating in this PR.

I think we should wait to update the domain tooltips on the option page until https://github.com/EFForg/privacybadger/issues/3133 is fixed (although I haven't been able to replicate on my computer). I moved the helper functions into a common location so that it's easy when the option page is ready. 

Do you mean that the "Sign in with google" warning should be tabbable? I added `aria-hidden` so it's read as the description of the selected toggle, but not read again at the end of the DOM. I'll [need to remove](https://dequeuniversity.com/rules/axe/4.4/aria-hidden-focus) that if we make it tabbable. Separately, it may be unclear to keyboard and non-keyboard users that clicking on the tooltip closes it, since I don't think that's a common pattern (unlike click-outside-to-close). To make it more accessible, we may want to add an X button and make that tabbable.

ghostwords commented:
>Do you mean that the "Sign in with google" warning should be tabbable?

Yeah, exactly. While tabbing through the popup, the keyboard never focuses on the warning, so you can't dismiss it with the keyboard.
